lib/aws/http_client.ex

```diff
@@ -23,7 +23,14 @@ defmodule AWS.HTTPClient do
     body :: iodata(),
     headers :: list(),
     options :: keyword()
-  ) :: {:ok, %{status_code: integer(), body: map()}} | {:error, term()}
+  ) :: {:ok, %{status_code: integer(), headers: list(), body: map()}} | {:error, term()}
```

Comment:
`headers` is needed by https://github.com/aws-beam/aws-elixir/blob/master/lib/aws/request.ex#L191

Reply:
We could include the shape of the expected tuple here too. WDYT?

```suggestion
  ) :: {:ok, %{status_code: integer(), headers: list({binary(), binary()}), body: map()}} | {:error, term()}
```

Reply:
I like it! Should we do the same for the request headers?